lib/Backends/CPU/LLVMIRGen.cpp

```diff
@@ -1266,7 +1267,7 @@ void LLVMIRGen::generateLLVMIRForInstr(llvm::IRBuilder<> &builder,
     // this number output slices each iteration.
     unsigned unrollDFactor = 1;
 
-    if ((destDepth % 8) == 0) {
+    if ((destDepth % 8) == 0 && (destDepth / 8) % (CI->getGroup()) == 0) {
```

Comment:
Please make this part more readable. Split the expression into multiple boolean expressions and add comments. 

For example: 
bool groupDividedBy8 = ((destDepth / 8) % CI->getGroup()) == 0;